cmd/juju/commands/bootstrap.go

```diff
@@ -745,16 +740,15 @@ See `[1:] + "`juju kill-controller`" + `.`)
 		return errors.Annotate(err, "saving bootstrap endpoint address")
 	}
 
+	if err = c.SetModelName(modelcmd.JoinModelName(c.controllerName, c.hostedModelName), false); err != nil {
+		return errors.Trace(err)
+	}
+
 	if isCAASController {
 		// TODO(caas): wait/ping/ensure controller public endpoint is healthy/functioning(bootstrap state done, agent starts, ELB provisioned, etc).
```

Comment:
Let's add this back in now - the waitAgentInitialisation() should just work right?

Reply:
sure, now it waits api server is ready.